chips/cc26x0/src/udma.rs

```diff
@@ -0,0 +1,420 @@
+    dest_ptr: usize,
+    control: ReadWrite<u32,DMATableControl::Register>,
+    _unused: usize,
+}
+
+/*
+pub struct DMAChannel {
+    channel: DMAPeripheral,
+    client: Cell<Option<&'static DMAClient>>,
+    width: Cell<DMAWidth>,
+    enabled: Cell<bool>,
+    xfer_type: Cell<DMATransferType>
+}
+*/
+
+/*
+pub trait DMAClient {
+    fn xfer_done(&self, pid: DMAPeripheral);
+}
+*/
+
+pub static mut UDMA: Udma = Udma::new();
+
+static mut DMACTRLTAB: DMAChannelControlTable = DMAChannelControlTable::new();
+
+impl DMAChannelControlTable{
+    const fn new() -> DMAChannelControlTable{
+        DMAChannelControlTable{
+            config_array: [ 
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(), 
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(),
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(), 
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(),
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(), 
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(),
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(), 
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(),
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(), 
+                DMAChannelControl::new(), DMAChannelControl::new(), DMAChannelControl::new(),
+                DMAChannelControl::new(), DMAChannelControl::new()
+            ]
+        }
+    }
+}
+
+impl DMAChannelControl {
+    const fn new() -> DMAChannelControl {
+        DMAChannelControl {
+            source_ptr: 0, 
+            dest_ptr: 0, 
+            control: ReadWrite::new(0), 
+            _unused: 0 
+        }
+    }
+}
+
+impl Udma {
+    /// Constructor 
+    pub const fn new() -> Udma {
+        Udma {
+            regs: UDMA_BASE as *const DMARegisters
+        }
+    }
+
+    /// Enable Function
+    /// Sets up the power domain and the clocks
+    /// It then sets MASTERNEABLE in the CFG register and writes the location of
+    /// the control table in the CTRL register.
+    pub fn enable(
+        &self,
+    ) {
+        let regs = unsafe{&*self.regs};
+        self.power_and_clock();
+        regs.cfg.write(Config::MASTERENABLE::SET);
+        unsafe{
+            regs.ctrl.set(&mut DMACTRLTAB as *mut DMAChannelControlTable as u32)
+        }
+    }
+
+    fn power_and_clock(&self) {
+        prcm::Power::enable_domain(prcm::PowerDomain::Peripherals);
+        while !prcm::Power::is_enabled(prcm::PowerDomain::Peripherals) {}
+        prcm::Clock::enable_dma();
+    }
+
+    pub fn initialize_channel(
+        &self,
+        dma_channel: DMAPeripheral, 
+        width: DMAWidth, 
+        xfer_type: DMATransferType,
```

Comment:
Please use more descriptive variable names if it's not too long, `transfer_type` is more explicit and clear than `xfer_type`